systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesIsolatedST.java

```diff
@@ -330,4 +333,72 @@ void testSwitchingStrimziPodSetFeatureGateOnAndOff(ExtensionContext extensionCon
 
         ClientUtils.waitForClientsSuccess(producerName, consumerName, INFRA_NAMESPACE, messageCount);
     }
+
+    /**
+     * UseKRaft feature gate
+     */
+    @IsolatedTest("Feature Gates test for enabled UseKRaft gate")
+    @Tag(INTERNAL_CLIENTS_USED)
+    public void testKRaftMode(ExtensionContext extensionContext) {
+        assumeFalse(Environment.isOlmInstall() || Environment.isHelmInstall());
+
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = "producer-test-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String consumerName = "consumer-test-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String topicName = KafkaTopicUtils.generateRandomNameOfTopic();
+
+        final LabelSelector zkSelector = KafkaResource.getLabelSelector(clusterName, KafkaResources.zookeeperStatefulSetName(clusterName));
+        final LabelSelector kafkaSelector = KafkaResource.getLabelSelector(clusterName, KafkaResources.kafkaStatefulSetName(clusterName));
+
+        int messageCount = 180;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 3;
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+UseStrimziPodSets,+UseKRaft", null));
+
+        clusterOperator.unInstall();
+        clusterOperator = new SetupClusterOperator.SetupClusterOperatorBuilder()
+                .withExtensionContext(BeforeAllOnce.getSharedExtensionContext())
+                .withNamespace(INFRA_NAMESPACE)
+                .withWatchingNamespaces(Constants.WATCH_ALL_NAMESPACES)
+                .withExtraEnvVars(testEnvVars)
+                .createInstallation()
+                .runInstallation();
+
+        Kafka kafka = KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build();
+        kafka.getSpec().setEntityOperator(null); // The builder cannot disable the EO. It has to be done this way.
+        resourceManager.createResource(extensionContext, kafka);
+
+        LOGGER.info("Try to send some messages to Kafka over next few minutes.");
+
+        KafkaClients kafkaBasicClientJob = new KafkaClientsBuilder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(messageCount)
+                .withDelayMs(500)
+                .withNamespaceName(INFRA_NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi());
+        JobUtils.waitForJobRunning(consumerName, INFRA_NAMESPACE);
```

Comment:
`JobUtils.waitForJobRunning(consumerName, INFRA_NAMESPACE);` is not needed as the `createResource` already waits for jobs to be running.